crates/symbolicator/src/server.rs

```diff
@@ -29,24 +39,46 @@ pub fn run(config: Config) -> Result<()> {
         .thread_stack_size(8 * megs)
         .build()?;
 
-    let socket = config.bind.parse::<SocketAddr>()?;
+    let mut servers: Vec<BoxFuture<_>> = vec![];
 
-    let service = web_pool
+    let service_http = web_pool
         .block_on(Service::create(
-            config,
+            config.clone(),
             io_pool.handle().to_owned(),
             cpu_pool.handle().to_owned(),
         ))
-        .context("failed to create service state")?;
+        .context("failed to create HTTP service state")?;
+    let socket_http = config.bind.parse::<SocketAddr>()?;
+    let server_http = axum_server::bind(socket_http)
+        .serve(endpoints::create_app(service_http).into_make_service());
+    servers.push(Box::pin(server_http));
+    tracing::info!("Starting HTTP server on {}", socket_http);
 
-    let _guard = web_pool.enter();
-    let server =
-        axum::Server::try_bind(&socket)?.serve(endpoints::create_app(service).into_make_service());
-
-    tracing::info!("Starting server on {}", server.local_addr());
-
-    web_pool.block_on(server)?;
+    if let Some(ref bind_str) = config.bind_https {
+        let https_conf = match config.server_config.https {
+            None => panic!("Need HTTPS config"),
+            Some(ref conf) => conf,
+        };
+        let service_https = web_pool
+            .block_on(Service::create(
```

Comment:
Please re-use the existing service, as that is meant to be a global singleton. Having two services would mean that you wouldn’t share caches properly, and some dedupe logic would be missing.

For example, starting a request via http and polling via https would fail. connections to external symbol servers would be duplicated, etc.

The Service is `Clone` so that should not be a problem.

Reply:
Thanks @Swatinem, fixed!